daphne/src/roles.rs

```diff
@@ -326,13 +329,29 @@ where
             return Err(DapAbort::InvalidProtocolVersion);
         }
 
+        if collect_req.query == Query::FixedSizeCurrentBatch {
+            // This is where we assign the current batch, and convert the
+            // Query::FixedSizeCurrentBatch into a Query::FixedSizeByBatchId.
+            //
+            // TODO(bhalleycf) Note that currently we are just looking at the
+            // head of the uncollected batch queue, so there is no parallelism
+            // possible for collectors on a given task.  To allow multiple
+            // batches for a task to be collected concurrently for the same task,
+            // we'd need a more complex DO state that allowed us have batch
+            // state go from unassigned -> in-progress -> complete.
+            collect_req.query = Query::FixedSizeByBatchId {
+                batch_id: self.current_batch(req.task_id()?).await?,
+            };
+        }
+
         // Ensure the batch boundaries are valid and that the batch doesn't overlap with previosuly
         // collected batches.
+        let batch_selector = BatchSelector::try_from(collect_req.query.clone())?;
```

Comment:
Can we avoid the `clone()` here?

Reply:
Maybe, but we end up needing to clone an Interval or a batch id anyway.  I suggest we leave this as it is until future optimization work, and then only optimize it if it is hot.